addons/stock_account/models/product.py

```diff
@@ -749,8 +749,13 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves, is_r
         candidates = stock_moves\
             .sudo()\
             .filtered(lambda m: is_returned == bool(m.origin_returned_move_id and sum(m.stock_valuation_layer_ids.mapped('quantity')) >= 0))\
-            .mapped('stock_valuation_layer_ids')\
-            .sorted()
+            .mapped('stock_valuation_layer_ids')
+
+        if len(candidates) > 1:
+            candidates = candidates.sorted()
+
+        if self._context.get('prefetch_candidates'):
+            candidates = candidates.with_prefetch(self._context.get('prefetch_candidates'))
```

Comment:
Hi @ticodoo. Can you I get your opinion on this? Will you be okay with this change? It's for performance optimization.